mne/viz/epochs.py

```diff
@@ -65,71 +67,252 @@ def plot_epochs_image(epochs, picks=None, sigma=0., vmin=None,
+        elif not callable(combine):
+            raise ValueError(
+                "`combine` must be None, a callable or one out of 'mean' "
+                "or 'gfp'. Got " + str(type(combine)))
+        for ch_type, picks_, name in zip(all_ch_types, all_picks, names):
+            if len(np.atleast_1d(picks_)) < 2:
+                raise ValueError("Cannot combine over only one sensor. "
+                                 "Consider using different values for "
+                                 "`picks` and/or `groupby`.")
+
+            this_data = combine(data[:, picks_, :])
+            evoked = epochs.average(picks=np.atleast_1d(picks_))
+            evoked.data = this_data.mean(0)[np.newaxis, :]
+            to_plot_list.append((this_data, ch_type, evoked,
+                                 type2name.get(name, name)))
+
+    return to_plot_list  # data, ch_type, evoked, name
+
+
+def _plot_epochs_image(data, sigma=0., vmin=None, vmax=None, colorbar=False,
+                       order=None, show=False, unit=None, cmap=None,
+                       fig=None, axes=None, overlay_times=None, scaling=None,
+                       evoked=None, title=None, draw_evoked=False):
+    """Helper function for plot_epochs_image/epochs.plot_image."""
+    from scipy import ndimage
+    import matplotlib.pyplot as plt
+
+    ### prepare fig and axes ###
     if axes is not None:
         if fig is not None:
             raise ValueError('Both figure and axes were passed, please'
                              'decide between the two.')
         from .utils import _validate_if_list_of_axes
-        oblig_len = 3 if colorbar else 2
+        oblig_len = 3
+        if not colorbar:
+            oblig_len -= 1
+        if not draw_evoked:
+            oblig_len -= 1
         _validate_if_list_of_axes(axes, obligatory_len=oblig_len)
-        ax1, ax2 = axes[:2]
+        ax1 = axes[0]
+        if draw_evoked:
+            ax2 = axes[1]
         # if axes were passed - we ignore fig param and get figure from axes
         fig = ax1.get_figure()
         if colorbar:
             ax3 = axes[-1]
-    evoked = epochs.average(picks)
-    data = epochs.get_data()[:, picks, :]
+    else:
+        if fig is None:
+            fig = plt.figure()
+        plt.figure(fig.number)
+        ax1 = plt.subplot2grid((3, 10), (0, 0), colspan=9 if colorbar else 10,
+                               rowspan=2 if draw_evoked else 3)
+        if draw_evoked:
+            ax2 = plt.subplot2grid((3, 10), (2, 0),
+                                   colspan=9 if colorbar else 10, rowspan=1)
+        if colorbar:
+            ax3 = plt.subplot2grid((3, 10), (0, 9), colspan=1, rowspan=3)
+
+    ### collect parameters about the data ###
     n_epochs = len(data)
-    data = np.swapaxes(data, 0, 1)
-    if sigma > 0.:
-        for k in range(len(picks)):
-            data[k, :] = ndimage.gaussian_filter1d(
-                data[k, :], sigma=sigma, axis=0)
```

Comment:
The smoothing was previously done here so that vmin and vmax are with respect to smoothed data (otherwise the images were often pale).